File: subs/convend/convend.f90
!c
!c  subroutine convend
!c  [history]
!c  2010/11/19 Yamashita
!c
!c=====
subroutine convend(outbuf, inbuf, bufsize)
  implicit none
  integer(4), intent(in) :: bufsize
  character(1), intent(in) :: inbuf(bufsize)
  character(1), intent(out) :: outbuf(bufsize)
  !c+++ work
  integer(4) :: i

  do i = 1, bufsize
    outbuf(i) = inbuf(bufsize-i+1)
  enddo

  return
end subroutine convend

File: convend/convend_c/convend.c

/* 2012/08/18 Yamashita */

int convend_(char buf[], char bufi[]){
  int i;
  int bufsize=sizeof(buf);
  for (i=0; i<=bufsize/2-1; i++){
    buf[i] = bufi[bufsize/2-i-1];
  }
}
